File: src/desktop/ui/src/Actions/Models/OperationErrorType.ts
export const operationErrorTypes = [
  "NotFound",
  "Validation",
  "MalformedInput",
  "UnprocessableInput",
  "Conflict",
  "InvalidOperation",
  "Unauthorized",
  "Timeout",
  "ExternalDependencyFailure",
  "InternalError",
] as const;

export type OperationErrorType = (typeof operationErrorTypes)[number];

export function isOperationErrorType(
  value: unknown,
): value is OperationErrorType {
  return operationErrorTypes.some((type) => type === value);
}

File: src/desktop/src-tauri/src/engine_protocol/models/operation_error_type.rs
use serde::{Deserialize, Serialize};

#[derive(Clone, Copy, Debug, Deserialize, PartialEq, Eq, Serialize)]
pub enum OperationErrorType {
    NotFound,
    Validation,
    MalformedInput,
    UnprocessableInput,
    Conflict,
    InvalidOperation,
    Unauthorized,
    Timeout,
    ExternalDependencyFailure,
    InternalError,
}
